### PR TITLE
V3

### PR DESCRIPTION
**Pull request by kamalmeh**
### Summary
1. 02.compile.pl is converted to 02.compile.ts
2. Changed the package.json
3. Script has code that works on both windows and linux operating systems
4. Tested only on windows as I have windows machine. I had to make this platform independent to work on windows.

### Testing
Below is the test carried out after compile step. Please note that just to execute compile step, I copied the 'en' directory in 'gu'. Then I executed below command shown in screenshot.
image

### CODE REVIEW

File: tsconfig.json
Comment (bibleutils) [inline tsconfig.json:1]: Could you use this tsconfig file:
```
{
	"compilerOptions": {
		"allowJs": true,
		"esModuleInterop": true,
		"moduleResolution": "node",
		"target": "esnext",
		"isolatedModules": true,
		"declaration": false,
		"allowSyntheticDefaultImports": true,
		"sourceMap": true,
		"resolveJsonModule": true,
		"strictNullChecks": false,
		"noImplicitAny": false,
		"lib": ["esnext", "dom"],
		"suppressExcessPropertyErrors": true,
		"types": [
			"node",
			"jest"
		]
	},
	"include": [
		"src/**/*",
		"src/**/*.json",
		"bin/**/*"
	],
	"exclude": [
		"node_modules"
	]

}
```

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ node_modules
 /temp_*
 /js/_bcv_parser.js
 .idea
-.DS_Store
```
Comment (bibleutils) [inline .gitignore:5]: Could you also exclude the `dist` directory?

**Comment by kamalmeh** [comment]
Sure. Thanks for letting me know.

Regards


Kamal Mehta
+91-9766320048

**Comment by bibleutils** [comment]
@kamalmeh thanks, 
Couple more things:

- Please remove the following files:

```
	deleted:    bin/01.add_lang.js
	deleted:    bin/02.compile.js
	deleted:    dist/01.add_lang.js
	deleted:    dist/01.add_lang.js.map
	deleted:    dist/02.compile.js
	deleted:    dist/02.compile.js.map
```
(They are just build files and don't need to be part of source control).
- Please install `tsx` (`npm i tsx`) and use it instead of ts-node
Screenshot 2024-09-06 at 6 10 58 PM

- Please reformat the 02.compile.ts file with the tab character (`\t`) instead of spaces

**Pull request merged**